### ISSUE TITLE
Test Working. Echo Dot not. 

### ISSUE THREAD

**Issue by EddWills95** [rgraciano/echo-sonos#77]
I can use both the Lambda and Alexa Skills page to test the echo sonos. 

For example I'll write 

> Ask Sonos .. "to play magic in the playroom" 

and it works fine after a few seconds. 

If I say exactly the same thing to the amazon echo it doesn't work and I get the "I'm not quite sure how to help you with that." So it appears that my echo dot doesn't want to connect to the skill. 

Any help would be great.

**Comment by gijoecool** [comment]
Is your skill name capitalized? I had a similar problem and it was because the skill Sonos was not capitalized.

**Comment by EddWills95** [comment]
The Skill is "Sonos" with a capital 'S' if that's what you mean.

**Comment by eyoung71** [comment]
You might open your app on your phone or device and see what it says when that command is issued.

**Comment by rgraciano** [comment]
@EddWills95, any luck?

**Comment by EddWills95** [comment]
not really. Still working in the testing stages. But my alexa dot, doesn't want to budge. 

@eyoung71 checking in the app it seems to be getting the right command: 

> ask sonos to play magic in the kitchen

for whatever reason the echo isn't even trying to connect to the skill ?

**Comment by jeffBonBass** [comment]
Hi - two things that gave me similar problems - first, I had to make sure the dot is attached to the same Amazon profile as the developer account; and second, I had a VPN active that prevented my dynamic dns setup from reaching my node server.

**Comment by jonyow** [comment]
Hi,

I've had the same issue:

- Testing works fine using the Service Simulator to send a request to Lambda for e.g. "Play Adele radio"
- Copying the sample JSON into the Lamdba test also works fine

- But when I try saying "Ask sonos to play adele radio", the Dot will just start playing a random radio station on it's own speaker.

- The Alexa history correctly picked up my command exactly as I said it. It looks like it decided to override my skill invocation with its own in build skill.

- I've also tried changing the invocation name to be a bit more verbose e.g. "Sonos music" and "Sonos speaker". Each time, Alexa hears the correct invocation name followed by the command, but still chooses to ignore the skill.

I'm based in the UK, if that makes any difference.

Any one experience something similar and found a solution?

Thanks!

**Comment by alasdairnicol** [comment]
I'm in the UK and have the same problem as @jonyow describes. I got it to work by creating a new skill with English (U.K.) as the language. Using English (U.K.) has its own problems, but #85 worked for me.

Perhaps adding an additional language to the original skill would work, I have not tested this approach, and it's time for bed now! If you do create a new skill, remember to update the appid in your options.js, create a new zip and re-upload to lambda.

**Comment by EddWills95** [comment]
I'll give that a go in the next week or so. I remember having the Amazon literal problem and probably just went with the US language. 

Seems to just be a strange issue for the UK. 

Thanks for all the responses.

**Comment by jonyow** [comment]
@alasdairnicol Thanks for the reply, that did the trick perfectly! 

And yes, I also had to use #85 to replace all the amazon.literals with a custom slot type, as English (U.K.) doesn't support amazon.literals.

**Comment by jonyow** [comment]
And thanks to @rgraciano for your awesome code and tutorial that made this possible. Now I don't have to anxiously wait until Sonos/Amazon release the official version :)

**Comment by EddWills95** [comment]
I'm still trying to get this to work. The Echo Dot doesn't do anything. 
I've recreated everything following the instructions from the original git and pull request from #85 
I'm now having problems with the Node package. Getting an error: 

` TypeError: Cannot read property 'roomName' of undefined`

May have made a fool of myself with something simple that i've missed.

**Comment by Zmav** [comment]
Did you get this working, I've just set this up in UK on an echo dot, and almost everything is working perfectly so if i can help ill try

**Comment by EddWills95** [comment]
I haven't tried in a while. Been very busy of late. will get back to you when I get some time. Thanks all for the replies. 

@Zmav is there a way I can get a hold of you as and when? Not sure if GitHub has a messaging facility.

**Comment by DirtySurti** [comment]
Ah one to try then. All works except with my dot. I am in UK and language is as English US. Will change that and see if works

**Comment by DirtySurti** [comment]
it worked !!!

**Comment by alokat7r** [comment]
I have solved the problem by selecting the same region my aws account is for lambda & alexa skill app
